robottelo/host_helpers/cli_factory.py

```diff
@@ -0,0 +1,285 @@
+    :raise robottelo.cli.factory.CLIFactoryError: Raise an exception if object
+        cannot be created.
+    :rtype: dict
+    :return: A dictionary representing the newly created resource.
+
+    """
+    if not values:
+        values = {}
+    update_dictionary(options, values)
+    if credentials:
+        cli_object = cli_object.with_user(*credentials)
+    try:
+        result = cli_object.create(options)
+    except CLIReturnCodeError as err:
+        # If the object is not created, raise exception, stop the show.
+        raise CLIFactoryError(
+            'Failed to create {} with data:\n{}\n{}'.format(
+                cli_object.__name__, pprint.pformat(options, indent=2), err.msg
+            )
+        )
+    # Sometimes we get a list with a dictionary and not
+    # a dictionary.
+    if type(result) is list and len(result) > 0:
+        result = result[0]
+    return result
+
+
+ENTITY_FIELDS = {
+    'activation_key': {
+        'name': gen_alphanumeric,
+    },
+    'architecture': {
+        'name': gen_alphanumeric,
+    },
+    'content_view': {'_redirect': 'content_view_with_credentials'},
+    'content_view_with_credentials': {
+        '_entity_cls': 'ContentView',
+        'name': gen_alpha,
+    },
+    'content_view_filter': {
+        'name': gen_alpha,
+    },
+    'content_view_filter_rule': {},
+    'discoveryrule': {
+        'name': gen_alphanumeric,
+    },
+    'content_credential': {
+        'path': f'/tmp/{gen_alphanumeric()}',
+        'content-type': 'gpg_key',
+        'name': gen_alphanumeric,
+    },
+    'location': {
+        'name': gen_alphanumeric,
+    },
+    'model': {
+        'name': gen_alphanumeric,
+    },
+    'partition_table': {
+        'file': lambda: f'/tmp/{gen_alphanumeric()}',
+        'name': gen_alphanumeric,
+        'os-family': partial(gen_choice, constants.OPERATING_SYSTEMS),
+    },
+    'product': {'_redirect': 'product_with_credentials'},
+    'product_with_credentials': {
+        '_entity_cls': 'Product',
+        'description': gen_alpha,
+        'label': gen_alpha,
+        'name': gen_alpha,
+    },
+    'proxy': {'name': gen_alphanumeric},
+    'repository': {'_redirect': 'repository_with_credentials'},
+    'repository_with_credentials': {
+        '_entity_cls': 'Repository',
+        'name': gen_alpha,
+        'url': settings.repos.yum_1.url,
+    },
+    'role': {'name': gen_alphanumeric},
+    'filter': {},
+    'scap_policy': {
+        'name': lambda: gen_alphanumeric().lower(),
+    },
+    'subnet': {
+        'mask': gen_netmask,
+        'name': gen_alphanumeric,
+        'network': partial(gen_ipaddr, True),
+    },
+    'sync_plan': {
+        'description': gen_alpha,
+        'enabled': 'true',
+        'interval': lambda: random.choice(list(constants.SYNC_INTERVAL.values())),
+        'name': gen_alpha,
+        'sync-date': lambda: datetime.datetime.now().strftime('%Y-%m-%d %H:%M:%S'),
+    },
+    'host': {
```

Comment:
make_host requires just ip, mac, name, root-password, btw that is not enough
for make_fake_host which is somewhat host organization, location, domain, architecture, os, partition table and medium is required, how can I create make_fake_host ?